src/Logger.js

```diff
@@ -62,81 +63,109 @@ export default class Logger {
-    // https://github.com/winstonjs/winston/blob/bd06c4091e9e248e6fed6278e0699e5cfdc7de24/lib/winston/exception-handler.js#L94-L97
-    process.on('uncaughtException', this.onUnhandledException);
-
-    // Unhandled promise rejections should be treated as errors.
-    process.on('unhandledRejection', this.onUnhandledException);
   }
 
   createLogMethods() {
-    Object.keys(this.levels).forEach(level => {
-      this[level.toLowerCase()] = message => {
+    Object.keys(this.levels).forEach((level) => {
+      this[level.toLowerCase()] = (message) => {
         let meta;
-
         // Log functions can be passed an object to provide
         // additional meta data like a logger `name`.
         if (typeof message === 'object') {
           meta = {...message};
           delete meta.message;
           message = message.message;
         }
-
-        this.winston.log(level, message, meta);
+        // Pass the meta object as actual `meta` object
+        this.winston.log(level, message, {meta});
       };
     });
   }
 
-  onUnhandledException = e => {
+  exitOnError = (e) => {
     this.winston[this.getLevelsDescending()[0]](e.stack);
-    process.exit(1);
+    return true;
   };
 
   getLevelsDescending() {
     return Object.entries(this.levels)
       .sort((a, b) => b[1] - a[1])
-      .map(entry => entry[0]);
+      .map((entry) => entry[0]);
   }
 
-  formatJSONMessage(opts) {
-    const now = new Date();
-    const {name: logger_name, requestId, ...meta} = opts.meta || {};
-    const hasMeta = Object.keys(meta).length > 0;
-
-    return JSON.stringify({
-      service: this.service,
-      '@timestamp': now.toISOString(),
-      level: opts.level,
-      level_value: this.levels[opts.level],
-      logger_name,
-      message: opts.message,
-      meta: hasMeta ? meta : undefined,
-      stack_trace: hasMeta && meta.stack ? meta.stack.join('\n') : undefined,
-      requestId
-    });
+  format() {
+    const formats = [];
+
+    if (this.colorize && !this.isProduction) {
+      formats.push(this.formatColorize());
+    }
+
+    if (this.isProduction) {
+      formats.push(this.formatJson());
+    } else {
+      formats.push(this.formatSimple());
+    }
+
+    return winston.format.combine(...formats);
+  }
+
+  formatJson() {
+    return winston.format((info) => {
+      const now = new Date();
+
+      const {name: logger_name, requestId, ...meta} = info.meta || {};
+      const hasMeta = Object.keys(meta).length > 0;
+
+      const log = {
+        level: info.level,
+        message: info.message,
+        service: this.service,
+        '@timestamp': now.toISOString(),
+        level_value: this.levels[info.level],
+        logger_name,
+        meta: hasMeta ? meta : undefined,
+        stack_trace: hasMeta && meta.stack ? meta.stack.join('\n') : undefined,
+        requestId
+      };
+
+      info[Symbol.for('message')] = JSON.stringify(log);
+
+      return info;
+    })();
+  }
+
+  formatSimple() {
```

Comment:
```suggestion
  formatInline() {
```

? "simple" is always such a broad term 🙂